WordPress/src/main/java/org/wordpress/android/ui/comments/CommentsListFragment.java

```diff
@@ -739,4 +636,148 @@ public void onDestroyActionMode(ActionMode mode) {
         }
     }
 
+    private CommentAdapter getAdapter() {
+        if (mAdapter == null) {
+            mAdapter = new CommentAdapter(getActivity(), WordPress.getCurrentLocalTableBlogId());
+            mAdapter.setInitialState(mCommentAdapterState);
+            mAdapter.setOnCommentPressedListener(this);
+            mAdapter.setOnDataLoadedListener(this);
+            mAdapter.setOnLoadMoreListener(this);
+            mAdapter.setOnSelectedItemsChangeListener(this);
+        }
+
+        return mAdapter;
+    }
+
+
+    // adapter calls this when selected comments have changed (CAB)
+    @Override
+    public void onSelectedItemsChanged() {
+        if (mActionMode != null) {
+            if (getSelectedCommentCount() == 0) {
+                mActionMode.finish();
+            } else {
+                updateActionModeTitle();
+                // must invalidate to ensure onPrepareActionMode is called
+                mActionMode.invalidate();
+            }
+        }
+    }
+
+    @Override
+    public void onCommentPressed(int position, View view) {
+        Comment comment = getAdapter().getItem(position);
+        if (comment == null) {
+            return;
+        }
+        if (mActionMode == null) {
+            if (!getAdapter().isModeratingCommentId(comment.commentID)) {
+                mFilteredCommentsView.invalidate();
+                if (getActivity() instanceof OnCommentSelectedListener) {
+                    ((OnCommentSelectedListener) getActivity()).onCommentSelected(comment.commentID);
+                }
+            }
+        } else {
+            getAdapter().toggleItemSelected(position, view);
+        }
+    }
+
+    @Override
+    public void onCommentLongPressed(int position, View view) {
+        // enable CAB if it's not already enabled
+        if (mActionMode == null) {
+            if (getActivity() instanceof AppCompatActivity) {
+                ((AppCompatActivity) getActivity()).startSupportActionMode(new ActionModeCallback());
+                getAdapter().setEnableSelection(true);
+                getAdapter().setItemSelected(position, true, view);
+            }
+        } else {
+            getAdapter().toggleItemSelected(position, view);
+        }
+    }
+
+    private boolean shouldRestoreCab() {
+        return hasAdapter() && !getAdapter().getSelectedCommentsId().isEmpty() && mActionMode == null;
+    }
+
+    private void restoreCab() {
+        if (getActivity() instanceof AppCompatActivity) {
+            ((AppCompatActivity) getActivity()).startSupportActionMode(new ActionModeCallback());
+            updateActionModeTitle();
+        }
+    }
+
+    @Override
+    public void onStart() {
+        super.onStart();
+        EventBus.getDefault().register(this);
+    }
+
+    @Override
+    public void onStop() {
+        EventBus.getDefault().unregister(this);
+        super.onStop();
+    }
+
+    public void onEventMainThread(CommentEvents.CommentModerationFinishedEvent event) {
```

Comment:
Although this isn't really necessary, I like to add `@SuppressWarnings("unused")` to all `onEvent` handlers to avoid the AS lint warning about it being unused.